### PR TITLE
[Vue] Documentation on the asset pipeline and how Vue/TypeScript are handled

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Refs https://github.com/matomo-org/matomo/pull/17940

This also removes some docs on UIControl which has been deprecated since AngularJS was introduced.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
@sgiehl / @tsteur can you look at and merge this PR too?

**Comment by sgiehl** [comment]
@tsteur might it make sense to let every core and maybe plugin developer have a look at that to check if everything is clear or any questions come up? For me everything sounds clear, but I'm quite deep into everything, so maybe some fresh eyes might see that differently...

**Comment by tsteur** [comment]
sounds good. @peterhashair @bx80 @justinvelluppillai  could you also have a read here to see if all looks good?

File: docs/4.x/getting-started-part-1.md [hunk 1]
```diff
@@ -69,6 +69,12 @@ The following tools aren't required for this guide, but you may find them useful
 - **[xhprof](https://github.com/facebook/xhprof)** If you'd like to profile your code and debug any inefficiencies.
 - **[python](https://www.python.org/)** If you want to use the log importer.
 
+**Important Note for Windows**
+
+It is possible to develop Matomo On Windows, but some things will not work as well. Specifically, Windows users
+coding with Vue may encounter problems with compiling Vue code. This is currently an unsolved problem, and we'd happily
+accept contributions that would help Windows users have an easier time contributing to Matomo.
+
```
Comment (justinvelluppillai) [inline docs/4.x/getting-started-part-1.md:77]: can you link an issue or provide more description of the problems on windows?

**Comment by justinvelluppillai** [comment]
Looks reasonably clear to me.

File: docs/4.x/getting-started-part-1.md (hunk 1 above)
Comment (diosmosis) [reply docs/4.x/getting-started-part-1.md:77]: there's no issue, but I can create one. There might be a lot of problems, I just gave up at the one that I couldn't fix :) . 

This problem being that the typescript compiler kept complaining about about relative imports (eg, `import ... from './ExampleComponent.vue`). The error was a bit vague, and from googling it seems like many people have this issue on windows w/ typescript.

Comment (diosmosis) [reply docs/4.x/getting-started-part-1.md:77]: this was the error that stopped me:

image

Comment (diosmosis) [reply docs/4.x/getting-started-part-1.md:77]: This is the issue that made me decide to stop working on it: https://github.com/vuejs/vue-cli/issues/2934

I'll put all of this in a new issue tomorrow.

**Review by sgiehl — APPROVED** [review]
Looks clear to me. If no one else has any question, guess this should be good to merge.

**Comment by peterhashair** [comment]
Looks clear to me

File: docs/4.x/getting-started-part-1.md (hunk 1 above)
Comment (diosmosis) [reply docs/4.x/getting-started-part-1.md:77]: Created https://github.com/matomo-org/matomo/issues/18081

**Pull request merged**